### ISSUE TITLE
Better(?) data source for bavaria(germany)

### ISSUE THREAD

**Issue by arnesetzer** [whosonfirst-data/whosonfirst-data#2032]
Hi guys,

since the german data source (bkg) is not bad, there are some inaccuracies due to the scale of the data (1:250.000). There are better data sources from the official bavarian state office for digitisation, broadband and measurement ( Landesamt für Digitalisierung, Breitband und Vermessung) under the CC BY 4.0 license (https://www.ldbv.bayern.de/produkte/opendata/opendata_weitere.html https://www-ldbv-bayern-de.translate.goog/produkte/opendata/opendata_weitere.html?_x_tr_sl=auto&_x_tr_tl=de&_x_tr_hl=de&_x_tr_pto=wapp). 

Gemeinden: Locality
Verwaltungsgemeinschaften: localadmin
Landkreise: County
Regierungsbezirke: macrocounty
Bayern: region
Verwaltungsbezirke: All above in one file

I'm a little bit overstrained with the whos on first structure so I need help to add these data into whos on first. 

These data are in UTM32 (EPSG 25832) so they need to be transformed into WGS84 (EPSG 4326) and the .geojson format right?
That's done [here](https://cloud.setzer.one/s/MTYigkC95g4DXTg).

I made a PR for the data sources. Can be found: https://github.com/whosonfirst/whosonfirst-sources/pull/203

What needs to be done next?

**Comment by stepps00** [comment]
This is great, thanks @arnesetzer 

> These data are in UTM32 (EPSG 25832) so they need to be transformed into WGS84 (EPSG 4326) and the .geojson format right?

Yes, the geometry will need to be transformed before it can be imported into Who's On First. We'd also want to remove the z coord in the geometry so the geometry only contains x/y coords.

> What needs to be done next?

The first (and most time consuming step) would be to add an attribute to each feature in the source data containing the corresponding wof:id for each. So, for the Bayern_bayern.geojson file, we'd want to add a property with [`85682571`](https://spelunker.whosonfirst.org/id/85682571/) as the value. That way, we know which WOF feature each of the new geometries will be assigned to.

Then, we'd want to review the attributes in each of these geojson files to see if we can adjust any properties in the WOF feature(s). If the `de-ldbv` source maintains a unique ID for each feature, we could add that ID as a concordance `de-ldbv:id` concordance in the WOF records' `wof:concordances` properties, for example.

After those steps are complete, we could review the changes and craft a few scripts to automate the import into WOF. Happy to help here, let us know if there are any questions.

**Comment by arnesetzer** [comment]
Hi, thanks for the fast reply. Indeed I need some help. I'm adding the `wof:id` to the files and found out that there are two files for the same county. If you may have a look at https://github.com/whosonfirst-data/whosonfirst-data/issues/2035 and resolve it would really help me.

**Comment by stepps00** [comment]
It looks like in each of the reported issues with German admin places (#2034, #2035, and #2036), there are some correctly set `mz:` properties in the "duplicate" record that should be used to indicate whether or not the place(s) should be included in a hierarchy/address string, or indicate the place(s) are no longer current.

The localadmin placetype is used throughout Germany, which generally mimics the county placetype IIRC.. but in these cases, the county will have a `mz:hierarchy_label` property set to 0. The dataset you're working with may have features that map to more than one WOF feature, which is okay, since the WOF placetype hierarchy might duplicate features and use `mz:` properties to flag which features a dowstream consumer of WOF should rely on.

tldr: in these cases, we should maintain the placetype structure as-is, but map more than one wof:id to the source geometry.

**Comment by arnesetzer** [comment]
FTR: I added for each feature a key `wof:id` with the value of the ID that should be overwritten. 

Now the attributes


| Attribute name | example value | description |
|---|---|---|
| "Name" | "Freistaat Bayern" | Full Name, already included in the |
| "description" | `"Freistaat Bayern<center><table><tr><th colspan='2' align='center'><em>Attributes</em></th></tr><tr bgcolor=\"#E3E3F3\">\n<th>LAND</th>\n<td>BY</td>\n</tr><tr bgcolor=\"\">\n<th>MODELLART</th>\n<td>Basis-DLM#DTK25</td>\n</tr><tr bgcolor=\"#E3E3F3\">\n<th>OBJART</th>\n<td>75005</td>\n</tr><tr bgcolor=\"\">\n<th>OBJART_TXT</th>\n<td>AX_Gebiet_Bundesland</td>\n</tr><tr bgcolor=\"#E3E3F3\">\n<th>OBJID</th>\n<td>DEBYBDLMjK0000nf</td>\n</tr><tr bgcolor=\"\">\n<th>BEGINN</th>\n<td>2023-02-16T21:13:57Z</td>\n</tr><tr bgcolor=\"#E3E3F3\">\n<th>ADM</th>\n<td>2002</td>\n</tr><tr bgcolor=\"\">\n<th>BEZ_LAN</th>\n<td>Freistaat Bayern</td>\n</tr><tr bgcolor=\"#E3E3F3\">\n<th>SCH</th>\n<td>09</td>\n</tr></table></center>"` | Same like the stuff below, but in HTML |
| "timestamp" | null | Always null |
| "begin" | null | Always null |
| "end" | null | Always null |
| "altitudeMode" | "clampToGround" | Artefact from the kml => geojson conversion https://developers.google.com/kml/documentation/altitudemode?hl=de |
| "tessellate" | 1 | Artefact from the kml => geojson conversion https://developers.google.com/kml/documentation/kmlreference?hl=de#linearring |
| "extrude" | 0 | Artefact from the kml => geojson conversion https://developers.google.com/kml/documentation/kmlreference?hl=de#linearring |
| "visibility" | 1 | ? |
| "drawOrder" | null | Artefact from the kml always null |
| "icon" | null | Always null |
| "LAND" | "BY" | Region |
| "MODELLART" | "Basis-DLM#DTK25" | Type of source "basic digital  |
| "OBJART" | "75005" | Internal key for object definition https://www.ldbv.bayern.de/file/pdf/8478/ATKIS-OK%20Basis-DLM%206_0.pdf#page=319 |
| "OBJART_TXT" | "AX_Gebiet_Bundesland" | Object type as text |
| "OBJID" | "DEBYBDLMjK0000nf" | |
| "HDU_X" | null | always null |
| "BEGINN" | "2023-02-16T21:13:57Z" | start of the lifetime interval (I guess its here the creation date) |
| "ENDE" | null | start of the lifetime interval, always null |
| "ADM" | "2002" | Internal key for area definition. https://www.ldbv.bayern.de/file/pdf/8478/ATKIS-OK%20Basis-DLM%206_0.pdf#page=301 |
| "AVG" | null | If part of a [Verwaltungsmeinschaft](https://de.wikipedia.org/wiki/Verwaltungsgemeinschaft_(Bayern)) there would be the ID here. Is nowhere the case here |
| "BEZ_GEM" | null | Locality or null |
| "BEZ_KRS" | null | county or null |
| "BEZ_LAN" | "Freistaat Bayern" | region, always "Freistaat Bayern" |
| "BEZ_RBZ" | null | macrocounty or null |
| "SCH" | "09"| Offical german system to dertermine all localities in germany [AGS](https://de.wikipedia.org/wiki/Amtlicher_Gemeindeschl%C3%BCssel) up to 8 digits filled up with trailing zeros |


I would only keep "SCH" since it is an official unique id.

**Comment by arnesetzer** [comment]
How do I update the geometries? I found [this](https://github.com/whosonfirst/whosonfirst-cookbook/blob/main/how_to/fixing_geometries.md) and [this](https://github.com/whosonfirst/whosonfirst-cookbook/blob/main/scripts/1550_simple_geom_update.py) script which could do the job. But both of these scripts require the module `mapzen`, which I wasn't able to find anywhere. 

So my specific questions are:
- Did I select the correct scripts?
- Where do I find the mapzen module?
- How do I tell the simple_geom_update script which files it should update? Is it simple as `-o /path/to/WoF-Repo/data`?

**Comment by CeallachKN** [comment]
how can we help to move this forward? Or is this on hold as long as the NUTS-imports are being evaluated?
For clarification, I'm working on the same project @arnesetzer does, so we would directly benefit from this.
I guess this requires deeper involvement in the projects Update / QM-Mechanisms, but I would be happy to help anyway.

**Comment by nvkelso** [comment]
Hey @CeallachKN and @arnesetzer, sorry we lost track of this issue. We didn't realize you were blocked. @stepps00 can you provide guidance, please?

Your Bavaria project isn't blocked on NUTS imports.

For your project, is there a better source of data besides a KML file?

The imported packages in those scripts can be found in repos like:

- https://github.com/whosonfirst/py-mapzen-whosonfirst-utils
- https://github.com/whosonfirst/py-mapzen-whosonfirst-placetypes
- https://github.com/whosonfirst/py-mapzen-whosonfirst-export

**Comment by CeallachKN** [comment]
Not really blocked but we are in the final stages of replacing the Geocoding we use  with Pelias and recognized some artifacts (Adresses/Venues etc. not in the correct locality, since locality-borders cut through the settlements), due to the scale of the geometries (1:250.000 generalization) in WOF right now and since data with a scale of 1:25.000 is provided by the responsible governmental body based on Cadastral Data directly managed by them we thought, that we could help the project as well as us.

I'm not sure what you mean with better source of data besides a KML File but the LDBV-Bayern provided ZIP-Files contain KML as well as Shapefiles and @arnesetzer already converted these to geojson and reprojected them to EPSG:4326

> These data are in UTM32 (EPSG 25832) so they need to be transformed into WGS84 (EPSG 4326) and the .geojson format right?
That's done [here](https://cloud.setzer.one/s/MTYigkC95g4DXTg).

![grafik](https://github.com/whosonfirst-data/whosonfirst-data/assets/68635133/1c1401a8-11fb-4953-8f8d-3dfc75814136)
blue is WOF-Data
red is the LDBV-data at 1:25.000 scale

**Comment by nvkelso** [comment]
The new data looks great! I think we're pretty close here now...

Let us know if you need more help after trying those packages and the script!

**Comment by arnesetzer** [comment]
As mentioned above, instructions on how to overwrite the geometries would be nice. Also an estimation of which original data should be taken over would be great. I would keep "SCH", since it is the offical key to pin point a administrative area in german authorities and "OBJID" because thats the interal key.

**Comment by stepps00** [comment]
@arnesetzer @CeallachKN apologies for the late reply here. I just uploaded a script that you should be able to repurpose for this issue: https://github.com/whosonfirst/whosonfirst-cookbook/commit/eeeb1680c62eb43a951cecd55785cc58d972cf0a

Passing a geojson file where each feature has an `id` (equal to a valid `wof:id`) will update geometries, properties, and alt files as needed for the places you'd like to update. As for the "SCH" key, I'd recommend that be added to each feature's `wof:concordances` property.
